apps/sharedtest/test/unit/crop_resize_rotate_test.js

```diff
@@ -4,13 +4,123 @@
+      reject(new Error('Failed' + xhr.status));
+    };
+    xhr.responseType = 'blob';
+    xhr.send();
+  });
+}
+
+suite('Metadata copy test', function() {
+  suiteSetup(function() {
+
+    // We need the mock lazy loader for jpeg-exif.
+    window.LazyLoader = window.MockLazyLoader;
+  });
+
+
+  [
+    // this one is horizontal
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: { width: 640, height: 480 } },
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetSize: null },
+    { fileName: '/test-data/blue.jpg', dims: true, rotated: false,
+      targetCrop: { left: 10, top: 10, width: 150, height: 100 },
+      targetSize: null },
+    // this one is vertical (Exif orientation)
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetSize: { width: 480, height: 640 } },
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetSize: null },
+    { fileName: '/test-data/red.jpg', dims: true, rotated: true,
+      targetCrop: { left: 10, top: 10, width: 100, height: 150 },
+      targetSize: null }
+  ].
+    forEach(function (testFile) {
+      var testName = 'Testing metadata copy for ' + testFile.fileName;
+      if (testFile.targetSize) {
+        testName += ' resized to ' + testFile.targetSize.width +
+          'x' + testFile.targetSize.height;
+      }
+      test(testName, function(done) {
+
+        fetchBlob(testFile.fileName).then(function (blob) {
+          assert.ok(blob);
+
+          var targetSize = testFile.targetSize;
+          var targetCrop = testFile.targetCrop === undefined ? null :
+              testFile.targetCrop;
+          cropResizeRotate(
+            blob, targetCrop, targetSize, 'image/jpeg+exif',
+            function (error, outBlob) {
+              assert.equal(error, null);
+              assert.ok(outBlob);
+
+              JPEGParser.readExifMetaData(
+                outBlob, function (error, metaData) {
+                  assert.ok(metaData);
+                  // check metaData
+                  assert.equal(metaData.Orientation, 1);
+
+                  if (testFile.dims) {
+
+                    var expectedX, expectedY;
+                    if (targetSize) {
+                      expectedX = targetSize.width;
+                      expectedY = targetSize.height;
+                    } else if (targetCrop) {
+                      expectedX = targetCrop.width;
+                      expectedY = targetCrop.height;
+                    } else {
+                      if (!testFile.rotated) {
+                        expectedX = 1280;
+                        expectedY = 960;
+                      } else {
+                        expectedX = 960;
+                        expectedY = 1280;
+                      }
+                    }
+
+                    assert.equal(metaData.PixelXDimension, expectedX);
+                    assert.equal(metaData.PixelYDimension, expectedY);
```

Comment:
It would be a stronger test if you also verified that some of the other EXIF data is preserved. Date, focal length, flash, etc. Maybe the test case objects in the array above could each have a few pieces of EXIF metadata to check.